install/kubernetes/cilium/templates/validate.yaml

```diff
@@ -9,3 +9,10 @@
     {{ fail "Hubble Relay requires .Values.hubble.enabled=true" }}
   {{- end }}
 {{- end }}
+
+{{/* validate service monitoring CRDs */}}
+{{- if not (.Capabilities.APIVersions.Has "monitoring.coreos.com/v1") }}
+  {{- if and (.Values.prometheus.enabled) (or (.Values.prometheus.serviceMonitor.enabled) (.Values.operator.prometheus.serviceMonitor.enabled)) }}
```

Comment:
nit: This feels more readable to me if we reverse the two checks. I.e., only check APIVersions for "monitoring.coreos.com/v1" if `serviceMonitor` is enabled. What do you think?

Also, for my understanding: what is the reason that we need to check for `.Values.prometheus.enabled`? Is it because if this is not enabled, then accessing `.Values.prometheus.serviceMonitor.enabled` will cause an error?

Reply:
> This feels more readable to me if we reverse the two checks. I.e., only check APIVersions for "monitoring.coreos.com/v1" if serviceMonitor is enabled. What do you think?

Good point, it makes sense to reverse two checks (e.g. static first, then runtime later) :100: 

> Also, for my understanding: what is the reason that we need to check for .Values.prometheus.enabled? Is it because if this is not enabled, then accessing .Values.prometheus.serviceMonitor.enabled will cause an error?

ServiceMonitoring CRDs is only created if .Values.prometheus.enabled is true right now, hence I just want to keep the same check.

Reply:
> ServiceMonitoring CRDs is only created if .Values.prometheus.enabled is true right now, hence I just want to keep the same check.

Makes perfect sense! Thanks!